### ISSUE TITLE
[Feature Request] add string types for `ms` package

### ISSUE THREAD

**Issue by akamos** [temporalio/sdk-typescript#1066]
### Is your feature request related to a problem? Please describe.

When for example using the `sleep` method, it's possible to describe the time as a string, because of the built-in `ms` package. It would be great to use the string types from `ms` instead of a simple `string` type, to have typechecking on the given time argument.

### Describe the solution you'd like

The solution would be to use the typing provided by `ms`: https://github.com/vercel/ms/blob/78ce59eab01c197b9133bf9752ae01fbc11f4976/src/index.ts#L44

**Comment by mjameswh** [comment]
@bergundy This is what I have in mind for this.

It maintains backward compatibility, at the cost of complaining about deprecation rather than entirely refusing incorrect syntaxes. We could encourage people to progressively switch to `Duration` instead of `number | string` in their own code, and schedule removal of the deprecated `number | string` form in a few months.

```
type Duration = number | MsFormatedValue;

export function sleep(ms: Duration): Promise<void>;

/** @deprecated */
export function sleep(ms: string): Promise<void>;

export function sleep(ms: number | Duration | string): Promise<void> {
  // ...
}
```

**Comment by bergundy** [comment]
Hmm.. the only thing I'm worried about here is users reading these values from external configurations so we'll have to figure out what we want to do there but I think your suggestion makes total sense @mjameswh.

**Comment by mjameswh** [comment]
Fixed by #1136.
